packages/puppeteer-core/src/common/bidi/BrowsingContext.ts

```diff
@@ -240,34 +240,39 @@ export class BrowsingContext extends EventEmitter {
       timeout = this.#timeoutSettings.navigationTimeout(),
     } = options;
 
-    const waitUntilCommand = lifeCycleToSubscribedEvent.get(
+    const waitUntilEvent = lifeCycleToSubscribedEvent.get(
       getWaitUntilSingle(waitUntil)
     ) as string;
 
     await Promise.all([
       setPageContent(this, html),
       waitWithTimeout(
         new Promise<void>(resolve => {
-          this.once(waitUntilCommand, () => {
+          this.once(waitUntilEvent, () => {
             resolve();
           });
         }),
-        waitUntilCommand,
+        waitUntilEvent,
         timeout
       ),
     ]);
   }
 
   async content(): Promise<string> {
     return await this.evaluate(() => {
-      let retVal = '';
-      if (document.doctype) {
-        retVal = new XMLSerializer().serializeToString(document.doctype);
+      let content = '';
+      for (const node of document.childNodes) {
```

Comment:
can we deduplicate this or not yet?

Reply:
We can extract it as it the same way we did for `setContent`.

Reply:
let's do that!